### PR TITLE
Pmr docs

### PR DESCRIPTION
**Pull request by sdkrystian**

### CODE REVIEW

File: doc/qbk/02_6_allocators.qbk [hunk 1]
```diff
@@ -23,19 +23,13 @@ library's polymorphic allocator but with the following additional features:
   the managed object to be extended.
 
 * An implementation can indicate to the library that deallocation
-  is not necessary, improving the performance of destructors.
+  is not necessary, allowing destructor calls to be elided in certain situations.
```
Comment (vinniefalco) [inline doc/qbk/02_6_allocators.qbk:26]: "elided" yuck, use normal words! "allowing calls to destructors for elements of containers to be skipped in most cases."

File: doc/qbk/02_6_allocators.qbk [hunk 2]
```diff
@@ -54,52 +48,51 @@ These types and functions are available:
         A smart pointer container through which a __memory_resource__
 		is managed and accessed.
     ]
+][
+    [__make_counted_resource__]
+    [
+        Function to construct a reference-counted smart pointer
```
Comment (vinniefalco) [inline doc/qbk/02_6_allocators.qbk:54]: This function returns...

File: doc/qbk/02_6_allocators.qbk [hunk 3]
```diff
@@ -0,0 +1,213 @@
+[/
+    Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+	Copyright (c) 2020 Krystian Stasiowski (sdkrystian@gmail.com)
```
Comment (vinniefalco) [inline doc/qbk/02_6_allocators.qbk:3]: TAB

**Comment by vinniefalco** [comment]
Please squash this down to one commit

**Comment by sdkrystian** [comment]
Merged with #94

**Pull request closed without merging**